docs/00-engineering/Php/drupal.md

```diff
@@ -0,0 +1,522 @@
+# Drupal Development
+
+## Drupal Standards
+
+Most Drupal 8 and beyond projects at Bixal use some version of [Bixal drupal-project](https://github.com/Bixal/drupal-project). The readme file in your project should be the first thing you do when on-boarding on to a new project. It should contain answers to most questions about how to set up the project and where to get resources. If there are addtional questions contact the project team in Microsoft teams for help.
+
+### Drupal-specific Coding Standards
+
+* We follow [Drupal Coding Standards](https://www.drupal.org/docs/develop/standards) which includes:
+  * [PHP](https://drupal.org/coding-standards)
+  * [CSS](https://www.drupal.org/docs/develop/standards/css)
+  * [JS](https://www.drupal.org/docs/develop/standards/javascript) coding standards.
+  * [Writing Secure Code in Drupal 7](https://www.drupal.org/docs/7/security/writing-secure-code)
+  * [Writing Secure Code in Drupal 8](https://www.drupal.org/docs/8/security/writing-secure-code-for-drupal-8)
+* We ensure and test our coding standards using the built in automation in projects. Please check the readme.md of the project you are working on. Most projects will have a [Makefile](https://github.com/Bixal/drupal-project/blob/8.x/Makefile) or Python Script which makes this a near trivial effort. Certain projects or lead may enforce this effort using githook. All projects will test for this in Bitbucket pipelines, Gitlab, Github, CircleCi, etc.
+  * All module functions (even little helper functions) should have the module prefix.
+  * Database tables should have the module prefix.
+  * Variables should have the module prefix.
+  * Custom code, when possible, should be developed as a standalone library that a module can include and integrate with Drupal. This should mean using class namespaces and best OOP practices, and at that point we can use an alternative more flexible autoloading approach, like the one provided by composer.
+* We always try to use Drupal functions where they exist. This helps with upgrades (among other things).
+  * Use `drupal_get_path();` to create a path to a file in your module directory.
```

Comment:
This isn't officially deprecated (yet), but it's no longer best practice. The new way to do this is: 
`$module_handler = \Drupal::service('module_handler');
$module_path = $module_handler->getModule('my_module')->getPath();`